transformers/test_data/shared_behaviors/shared_flip_bid_trigger_suite.go

```diff
@@ -0,0 +1,407 @@
+		} else {
+			columnVal = "0x" + test_data.RandomString(40)
+		}
+		repoVal = map[int]string{0: columnVal}
+	default:
+		err = errors.New("ValueType not implemented")
+	}
+	Expect(err).NotTo(HaveOccurred())
+	return repoVal, columnVal
+}
+
+type commonBidSnapshot struct {
+	AddressID   string `db:"address_id"`
+	BlockNumber string `db:"block_number"`
+	BidID       string `db:"bid_id"`
+	Guy         sql.NullString
+	Tic         sql.NullString
+	End         sql.NullString
+	Lot         sql.NullString
+	Bid         sql.NullString
+	Updated     string
+}
+
+type flipBidSnapshot struct {
+	commonBidSnapshot
+	Usr sql.NullString
+	Gal sql.NullString
+	Tab sql.NullString
+}
+
+func commonBidSnapshotFromValues(bidID int, blockNumber, addressID int64, updated string, bidValues map[string]interface{}) commonBidSnapshot {
+	parsedUpdated, parseErr := strconv.ParseInt(updated, 10, 64)
+	Expect(parseErr).NotTo(HaveOccurred())
+	packedValues := bidValues[mcdStorage.Packed].(map[int]string)
+
+	return commonBidSnapshot{
+		AddressID:   strconv.FormatInt(addressID, 10),
+		BlockNumber: strconv.FormatInt(blockNumber, 10),
+		BidID:       strconv.Itoa(bidID),
+		Guy:         test_helpers.GetValidNullString(packedValues[0]),
+		Tic:         test_helpers.GetValidNullString(packedValues[1]),
+		End:         test_helpers.GetValidNullString(packedValues[2]),
+		Lot:         test_helpers.GetValidNullString(bidValues[mcdStorage.BidLot].(string)),
+		Bid:         test_helpers.GetValidNullString(bidValues[mcdStorage.BidBid].(string)),
+		Updated:     FormatTimestamp(parsedUpdated),
+	}
+}
+
+func flipBidSnapshotFromValues(bidID int, blockNumber, addressID int64, updated string, bidValues map[string]interface{}) flipBidSnapshot {
+	return flipBidSnapshot{
+		commonBidSnapshot: commonBidSnapshotFromValues(bidID, blockNumber, addressID, updated, bidValues),
+		Usr:               test_helpers.GetValidNullString(bidValues[mcdStorage.BidUsr].(string)),
+		Gal:               test_helpers.GetValidNullString(bidValues[mcdStorage.BidGal].(string)),
+		Tab:               test_helpers.GetValidNullString(bidValues[mcdStorage.BidTab].(string)),
+	}
+}
+
+func assertBidSnapshot(actualBid, expectedBid commonBidSnapshot, expectedBlockNumber int64) {
+	Expect(actualBid.AddressID).To(Equal(expectedBid.AddressID))
+	Expect(actualBid.BlockNumber).To(Equal(strconv.FormatInt(expectedBlockNumber, 10)))
+	Expect(actualBid.BidID).To(Equal(expectedBid.BidID))
+	Expect(actualBid.Lot).To(Equal(expectedBid.Lot))
+	Expect(actualBid.Bid).To(Equal(expectedBid.Bid))
+	Expect(actualBid.Updated).To(Equal(expectedBid.Updated))
+}
+
+func assertFlipSnapshot(actualFlip, expectedFlip flipBidSnapshot, expectedBlockNumber int64) {
+	assertBidSnapshot(actualFlip.commonBidSnapshot, expectedFlip.commonBidSnapshot, expectedBlockNumber)
+	Expect(actualFlip.Usr).To(Equal(expectedFlip.Usr))
+	Expect(actualFlip.Gal).To(Equal(expectedFlip.Gal))
+	Expect(actualFlip.Tab).To(Equal(expectedFlip.Tab))
+}
```

Comment:
We already have datatypes/functions similar to these in `test_helpers`, but these are slightly different since they're designed to work with the `maker.flip`-style records from the DB, whereas the ones in `test_helpers` work with the `api.flip_state`-style types (returned by `api.get_flip`, etc.).

So I think it makes sense to have 2 distinct sets of helpers for 2 distinct types. Although, hopefully we'll eventually be able to expose the `flip` table directly instead of exposing its data via `api.get_flips` and `all_flips`, at which point we'd be able to delete the other set of helpers.

Reply:
Actually no, it doesn't look like we do. For some reason I thought I added stories like that at the same time that I added the ones about making better use of the ilk + urn trigger tables, but I guess not 🤔. I'll see to getting some into the backlog.